File: LeetCode/Easy/0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode merge = new ListNode(-1);
        ListNode current = merge;
        ListNode left = list1;
        ListNode right = list2;

        while(left != null && right != null){
            if(left.val < right.val){
                current.next = left;
                left = left.next;
            }else{
                current.next = right;
                right= right.next;
            }
            current = current.next;
        }


        current.next = (left != null) ? left : right;

        return merge.next;
    }
}

File: LeetCode/Easy/0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(list1, list2) {
    let left = list1;
    let right = list2;
    const dummy = new ListNode();
    let cur = dummy;
    while(left && right){
        if(left.val < right.val){
            cur.next = left;
            left = left.next;
        }else{
            cur.next = right;
            right = right.next;
        }
        cur = cur.next;
    }

    cur.next = left ? left : right;
    return dummy.next;
};